cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/kafka/Kafka08SimpleApiConsumer.java

```diff
@@ -238,13 +238,13 @@ public void destroy() {
   }
 
   @Override
-  protected void processMessage(String key, ByteBuffer payload, Emitter<StructuredRecord> emitter) {
+  protected void processMessage(String key, byte[] payload, Emitter<StructuredRecord> emitter) {
```

Comment:
Why change the type to `byte[]`. As far I as know, the Kafka message has payload in ByteBuffer, decoding ByteBuffer into byte[] almost always involve array creation and copying (because of the position and limit), while passing ByteBuffer around is pretty simple and the overhead is usually pretty small.